cmd/clusterctl/cmd/config_cluster.go

```diff
@@ -189,9 +190,30 @@ func runGetClusterTemplate(cmd *cobra.Command, name string) error {
 
 func templateListVariablesOutput(template client.Template) error {
 	if len(template.Variables()) > 0 {
-		fmt.Println("Variables:")
+		required := make([]string, 0)
+		notRequired := map[string]string{}
 		for _, v := range template.Variables() {
-			fmt.Printf("  - %s\n", v)
+			if v != nil {
+				if v.Required() {
+					required = append(required, v.Name())
+				} else {
+					notRequired[v.Name()] = v.DefaultValue()
+				}
+			}
+		}
+		sort.Strings(required)
+		if len(required) > 0 {
+			fmt.Println("Required Variables:")
+			for _, v := range required {
+				fmt.Printf("  - %s\n", v)
+			}
+		}
+		fmt.Println()
+		if len(notRequired) > 0 {
+			fmt.Println("Optional Variables:")
+			for k, val := range notRequired {
+				fmt.Printf("  - %s (defaults to \"%s\")\n", k, val)
```

Comment:
tried this locally, looks good but the default value looks weird when it's an empty string because it gets double quoted:

```
bin/clusterctl config cluster cluster --list-variables --infrastructure=azure:v0.4.12
Required Variables:
  - AZURE_CONTROL_PLANE_MACHINE_TYPE
  - AZURE_LOCATION
  - AZURE_NODE_MACHINE_TYPE
  - AZURE_SUBSCRIPTION_ID
  - CLUSTER_NAME
  - CONTROL_PLANE_MACHINE_COUNT
  - KUBERNETES_VERSION
  - WORKER_MACHINE_COUNT

Optional Variables:
  - AZURE_RESOURCE_GROUP (defaults to "${CLUSTER_NAME}")
  - AZURE_SSH_PUBLIC_KEY_B64 (defaults to """")
  - AZURE_VNET_NAME (defaults to "${CLUSTER_NAME}-vnet")
  ````